qiskit_nature/drivers/pyscfd/pyscfdriver.py

```diff
@@ -125,49 +302,386 @@ def _check_valid():
+            output = None
+            if logger.isEnabledFor(logging.DEBUG):
+                verbose = pylogger.INFO
+                file, output = tempfile.mkstemp(suffix=".log")
+                os.close(file)
+
+            self._mol = gto.Mole(
+                atom=atom,
+                unit=self._unit,
+                basis=self._basis,
+                max_memory=self._max_memory,
+                verbose=verbose,
+                output=output,
             )
-            charge = self.molecule.charge
-            spin = self.molecule.multiplicity - 1
-            units = self.molecule.units.value
+            self._mol.symmetry = False
+            self._mol.charge = self._charge
+            self._mol.spin = self._spin
+            self._mol.build(parse_arg=False)
+
+            if output is not None:
+                self._process_pyscf_log(output)
+                try:
+                    os.remove(output)
+                except Exception:  # pylint: disable=broad-except
+                    pass
+
+        except Exception as exc:
+            raise QiskitNatureError("Failed to build the PySCF Molecule object.") from exc
+
+    @staticmethod
+    def _check_molecule_format(val: str) -> Union[str, List[str]]:
+        """Ensures the molecule coordinates are in XYZ format.
+
+        This utility automatically converts a Z-matrix coordinate format into XYZ coordinates.
+
+        Args:
+            val: the atomic coordinates.
+
+        Raises:
+            QiskitNatureError: If the provided coordinate are badly formatted.
+
+        Returns:
+            The coordinates in XYZ format.
+        """
+        atoms = [x.strip() for x in val.split(";")]
+        if atoms is None or len(atoms) < 1:  # pylint: disable=len-as-condition
+            raise QiskitNatureError("Molecule format error: " + val)
+
+        # An xyz format has 4 parts in each atom, if not then do zmatrix convert
+        # Allows dummy atoms, using symbol 'X' in zmatrix format for coord computation to xyz
+        parts = [x.strip() for x in atoms[0].split(" ")]
+        if len(parts) != 4:
+            try:
+                newval = []
+                for entry in gto.mole.from_zmatrix(val):
+                    if entry[0].upper() != "X":
+                        newval.append(entry)
+                return newval
+            except Exception as exc:
+                raise QiskitNatureError("Failed to convert atom string: " + val) from exc
+
+        return val
+
+    def run_pyscf(self) -> None:
+        """Runs the PySCF calculation.
+
+        This method is part of the public interface to allow the user to easily overwrite it in a
+        subclass to further tailor the behavior to some specific use case.
+
+        Raises:
+            QiskitNatureError: If an invalid HF method type was supplied.
+        """
+        try:
+            hf_method_name = self._hf_method.upper()
+            hf_method_cls = getattr(scf, hf_method_name)
```

Comment:
I guess this works - but it ends up more obscure for me and could do with explanation for any future readers if you choose to want to keep this. Certainly the switch/case (ifs from before) would grow larger here but to me its clearer/cleaner - now the enum names I guess have to match the method names.

Reply:
I can add some explanatory comment. I will not revert it to the `if-elif-else` switches.
I will wait with this update until we have resolved the other discussion points above.

Reply:
Thats fine - if its documented - though personally I am not so keen on the direct couping of names in the enum - that currently has wider use in Nature, to names that match PySCF driver methods, though I understand RHF/ROHF and UHF are universal enough - maybe that's true of these KS methods too.